pkg/config/config.go

```diff
@@ -248,8 +267,33 @@ func CustomHTTPErrorHandler(err error, c echo.Context) {
 	}
 }
 
-func ConfigureEcho() *echo.Echo {
-	e := echo.New()
+// See: https://echo.labstack.com/middleware/prometheus/#skipping-certain-urls
+func metricsMiddlewareSkipper(ctx echo.Context) bool {
+	path := ctx.Request().URL.Path
+	switch {
+	case path == "/ping" || path == "/ping/":
+		return true
+	case path == "/metrics" || path == "/metrics/":
+		return true
+	}
+	pathItems := handler_utils.NewPathWithString(path).RemovePrefixes()
+	return pathItems.StartWithResources(
+		[]string{"ping"},
+		[]string{"repositories", "validate"},
+		[]string{"repository_parameters", "validate"},
+	)
```

Comment:
What is `StartWithResources` used for?

Reply:
It is used for matching the starting items of the path once it is removed the prefix items ([/beta]/api/<service-name>/<version>).

Examples of matching:
* For "/api/content-sources/ping/something-else" would return true as it would match with `[]string{"ping"}`.
* For "/api/content-sources/repositories/validate/" would return true.
* For "/api/content-sources/anyotherthing" would return false.
* For "/api/repositories" would return false.

Reply:
Thanks. A couple more questions/comments.

1. `/ping` is already being skipped earlier in the code, so does it still need to be skipped here?
2. There's no `repositories/validate` endpoint, so I don't think this line is needed.
3. Why skip `repository_parameters/validate`? If that gave a `5xx` error, wouldn't we want prometheus to count that?

Reply:
1. `ping` is skipped for enforceidentity wrapper middleware, but that skipper apply only to that middleware, is another middleware is before or after the chain, and the chain is not interrupted, the skip would not apply and it needs to check its own cases.
2. Thanks! I missed that.
3. At some point I recall it was mentioned to do not get metrics for the validation endpoints, but I don't recall the motivation for it; @rverdile @jlsherrill  if I am wrong, let me know and I remove the skipper for it.

Reply:
1. To clarify, `ping` is also skipped in the first case of the switch statement in `metricsMiddlewareSkipper`, right before `StartWithResources` is called. With this in mind, did you still mean to pass ping to `StartWithResources`?

3. Ah okay, then I'm not sure whether or not validation should be skipped. It's a question for @jlsherrill :).

Reply:
Sorry, I thought it was a concern about the other middleware skipper. It is correct the filter for `/ping` into the first switch; we have 2 different paths for ping `/ping` and `/api/content-sources/v1/ping`; the switch is skipping the first ping path, the StartWithResources is skipping the second ping path.

Reply:
2)  The purpose of skipping that particular endpoint for metrics is that we don't want to measure things like latency on that endpoint.  The latency would be higher because its reaching out to  the internet to grab the repomd.xml file.

Reply:
Do we also want to skip counting errors caused by reaching out to the internet? 

An alternative option is to filter out the endpoint when querying for the latency metrics i.e. `content_sources_http_status_histogram_bucket{le="0.01",path!="/api/content-sources/v1.0/repository_parameters/validate"}`.

Reply:
Ah yes, thats a better option! Lets do that :)  that may simplify some code as well.

Reply:
I have removed `repository_parameters/validate` route for the skipper, and I missed to remove the unexisting route `repositories/validate`. Thanks for the comments